opentelemetry/src/context.rs

```diff
@@ -346,15 +350,17 @@ impl fmt::Debug for Context {
 /// A guard that resets the current context to the prior context when dropped.
 #[allow(missing_debug_implementations)]
 pub struct ContextGuard {
-    previous_cx: Option<Context>,
-    // ensure this type is !Send as it relies on thread locals
+    // The position of the context in the stack. This is used to pop the context.
+    cx_pos: usize,
```

Comment:
This'll be 8 bytes on a 64 bit system, but nonetheless is it worth asserting that it's less than max on `attach` ? It'd be self-documenting (yes, we've considered this), and also make it super clear what's happened if the user does anything particularly heinous to blow through contexts.

Reply:
This is one of the undecided and not tested parts. We just need to decide on how large we want the context stack to be, and then we can adjust this type.

Reply:
@cijothomas any opinions? 

I reckon we'll need assertions / tests around the limits anyway. `usize` will be at least 16 bits. We could use `u16` so that it's constant width and 65k entries is still _a lot_.

Reply:
> We could use u16 so that it's constant width and 65k entries is still a lot.

This is plentiful, but wondering - In the event we hit the limit somewhere how do we communicate that to the caller of `Attach`? Is just internal log sufficient?

Reply:
I think that logging is the right option here, and the `attach` will become a *no-op*, and we return a special `ContextGuard` that is a *no-op* as well. We've effectively hit a hard limit and the application will not work as expected.

Reply:
> I think that logging is the right option here, and the `attach` will become a _no-op_, and we return a special `ContextGuard` that is a _no-op_ as well. We've effectively hit a hard limit and the application will not work as expected.

Agree with this approach.